utils/api.go

```diff
@@ -19,6 +19,10 @@ import (
 
 func CheckOrigin(r *http.Request, allowedOrigins string) bool {
 	origin := r.Header.Get("Origin")
+	if origin == "" {
```

Comment:
I believe this is OK because the browsers should always send an Origin header. This will allow non-browser applications access to websockets. 
This function is not used for HTTP CORS handling.